engineering/conventions-go.md

```diff
@@ -0,0 +1,50 @@
+
+# Go Development Conventions
+
+## Introduction
+
+This guide documents development conventions for Go at source{d}. Check [general conventions](conventions.md) for language-independent conventions that are also applicable for Go projects.
+
+## Supported Go Versions
+
+* Support latest two stable major versions of Go (e.g. 1.9.x and 1.10.x).
+  * Some of our projects may support more versions, specially when there is wide-adoption outside source{d} (e.g. [go-git](https://github.com/src-d/go-git) currently supports three).
+
+## Dependency Management
+
+* If your project is an application:
+  * Use [dep](https://github.com/golang/dep) to manage dependencyes.
+  * Check in the `vendor` directory into git.
+* If your project is a library:
+  * You may use [dep](https://github.com/golang/dep) to manage dependencyes if needed. If you do, do not commit `vendor` directory.
+
+## Build System
+
+* If your project requires fetching or building dependencies, implement that in the `dependencies` rule.
+* If binaries can be built, do so in the `packages` rule. Note that [src-d/ci](https://github.com/src-d/ci) provides that rule by default.
+
+## Testing
+
+* Use [testify](https://github.com/stretchr/testify).
+* Group your tests in [suites](https://github.com/stretchr/testify#suite-package) where relevant.
+
+## Executables
+
+* Put sources for your executable commands in `cmd/<your command>`.
```

Comment:
most of our go projects use `cli` directory for it, only some `cmd`.
`cli` examples:
https://github.com/src-d/enry
https://github.com/src-d/borges
https://github.com/src-d/go-git
https://github.com/bblfsh/bblfshd
https://github.com/bblfsh/sdk
https://github.com/src-d/code-annotation

Should we use `cmd` for new projects?

Reply:
Oh, I didn't check properly when writing this. [`cmd`](https://github.com/search?utf8=%E2%9C%93&q=language%3Ago+path%3Acmd+package+main&type=) seems more widely adopted than [`cli`](https://github.com/search?utf8=%E2%9C%93&q=language%3Ago+path%3Acli+package+main&type=) in the Go community. But unless we agree on this, status quo (`cli`) would win.

Reply:
source{d} is, in fact, a first place where I see commands in `cli` directory instead of `cmd`. I believe that Go convention is to call it `cmd`.

Reply:
I believe that `cli` is way more meaningful than just a generic command, actually `cmd` is the derived for the `cli`, anyways the reason because is called `cli` and not `cmd` is that collides with the command package, that is the pattern that we are using, where `cmd` is the proper name. 

I don't is a big deal and since it's already in multiple places, just keep using `cli`, unless someone wants to do a PR to all the repositories changing this.

Reply:
I will do this kind of PR with a great pleasure :)